libs/editor/egui_editor/src/galley.rs

```diff
@@ -0,0 +1,160 @@
+use crate::cursor_types::{DocByteOffset, DocCharOffset};
+use crate::editor::Editor;
+use crate::layout_job::{Annotation, LayoutJobInfo};
+use egui::epaint::text::cursor::Cursor;
+use egui::{Color32, Galley, Pos2, Rect, Sense, Stroke, Ui, Vec2};
+use std::ops::Range;
+use std::sync::Arc;
+
+// todo: maybe a nice VisualAppearence struct is in order?
```

Comment:
Is this a stale todo since `VisualAppearence` struct is a thing in theme?

Reply:
there is such a struct but this isn't included in it; I have changes I'm working on to address it though